tests/functional/aws-node-sdk/test/object/websiteHead.js

```diff
@@ -0,0 +1,595 @@
+// Date: Wed, 21 Dec 2016 20:33:00 GMT
+// Server: AmazonS3
+// Transfer-Encoding: chunked
+// x-amz-error-code: AccessDenied
+// x-amz-error-message: Access Denied
+// x-amz-id-2: HteVr4cPi9iddwUdAifVTxLZcidlh
+// e1yxO6mtr1lqaGRSW/lZlQMGLbIztWOTUcDP3vMA6PwpNE=
+// x-amz-request-id: FD4EA11DD16537F5
+
+// 2) If redirect rule, redirect with 301 status code.
+
+// 3) If success, respond with object's headers:
+
+// HTTP/1.1 200 OK
+// Content-Length: 314
+// Content-Type: text/html
+// Date: Wed, 21 Dec 2016 20:32:52 GMT
+// ETag: "6af6552e24eb4bdcc83f1b227dfc3dda"
+// Last-Modified: Fri, 02 Dec 2016 18:06:55 GMT
+// Server: AmazonS3
+// x-amz-id-2: o2wNnLjNiRhiL83ji34IeV5vLV/kXoIN+nzh41XE/cz7+
+// 8C5P3O3SIRj7PYNmhhJqE2NyDA2h5g=
+// x-amz-meta-somekey: mymetadata!!!
+// x-amz-request-id: 89BBCC95D4738024
+
+// 4) If there is a redirect rule for an error, and get such error,
+// redirect in accordance with rule:
+
+// Called head with prefix zzz and got 403 so satisfied conditions for redirect
+// HTTP/1.1 301 Moved Permanently
+// Content-Length: 0
+// Date: Thu, 22 Dec 2016 02:47:48 GMT
+// Location: http://whatever.com.s3-website-us-east-1.amazonaws.com/zzz
+// Server: AmazonS3
+// x-amz-id-2: 5HXWYfdLQS8ZRBZJcKV1cxqxgKMTJzg74rGl61kRLA
+// KX/MgqE4dZCJ4d9eF59Wbg/kza40cWcoA=
+// x-amz-request-id: 0073330F58C7137C
+
+
+describe('Head request on bucket website endpoint', () => {
+    it('should return 404 when no such bucket', done => {
+        const expectedHeaders = {
+            'x-amz-error-code': 'NoSuchBucket',
+            // Need arsenal fixed to remove period at the end
+            // so compatible with aws
+            'x-amz-error-message': 'The specified bucket does not exist.',
+        };
+        makeHeadRequest(404, expectedHeaders, undefined, done);
+    });
+
+    describe('with existing bucket', () => {
+        beforeEach(done => s3.createBucket({ Bucket: bucket }, done));
+
+        afterEach(done => s3.deleteBucket({ Bucket: bucket }, done));
+
+        it('should return 404 when no website configuration', done => {
+            const expectedHeaders = {
+                'x-amz-error-code': 'NoSuchWebsiteConfiguration',
+                'x-amz-error-message': 'The specified bucket does not ' +
+                    'have a website configuration',
+            };
+            makeHeadRequest(404, expectedHeaders, undefined, done);
+        });
+
+        describe('with existing configuration', () => {
+            beforeEach(done => {
+                const webConfig = new WebsiteConfigTester('index.html');
+                s3.putBucketWebsite({ Bucket: bucket,
+                    WebsiteConfiguration: webConfig }, err => {
+                    assert.strictEqual(err,
+                        null, `Found unexpected err ${err}`);
+                    s3.putObject({ Bucket: bucket, Key: 'index.html',
+                        ACL: 'public-read',
+                        Body: fs.readFileSync(path.join(__dirname,
```

Comment:
Would sending a read stream to `Body` be faster here? https://nodejs.org/api/fs.html#fs_fs_createreadstream_path_options